src/main/java/ui/components/issuepicker/IssuePickerDialog.java

```diff
@@ -0,0 +1,210 @@
+        GuiElement element = new GuiElement(issue, models.getLabelsOfIssue(issue), models.getMilestoneOfIssue(issue),
+                                            models.getAssigneeOfIssue(issue), models.getAuthorOfIssue(issue));
+        IssueCard card = new IssueCard(element, isSuggestedIssue(issue, currentSuggestion));
+        card.setOnMouseClicked(e -> handleIssueClick(issue, card));
+        return card;
+    }
+
+
+    // Event handling 
+
+    /**
+     * Updates state of the issue picker based on the entire query
+     */
+    private final void handleUserInput(String query) {
+        state = new IssuePickerState(allIssues, query.toLowerCase());
+        populatePanes(state);
+    }
+    
+    private void handleIssueClick(TurboIssue issue, IssueCard card) {
+        issuepickerQueryField.setDisable(true);
+        Platform.runLater(card::requestFocus);
+        state.updateSelectedIssues(issue);
+        populateSelectedIssues(state.getSelectedIssues());
+    }
+
+    // UI helper methods
+
+    private void createMainLayout() {
+        FXMLLoader loader = new FXMLLoader(UI.class.getResource(FXML_PATH));
+        loader.setController(this);
+        try {
+            issuepickerLayout = (VBox) loader.load();
+        } catch (IOException e) {
+            logger.error("Failure to load FXML. " + e.getMessage());
+            close();
+        }
+    }
+
+    private void createButtons() {
+        ButtonType confirmButtonType = new ButtonType("Confirm", ButtonData.OK_DONE);
+        getDialogPane().getButtonTypes().addAll(confirmButtonType, ButtonType.CANCEL);
+
+        // defines what happens when user confirms/presses enter
+        setResultConverter(dialogButton -> {
+            if (dialogButton.getButtonData() == ButtonData.OK_DONE) {
+                // Confirms final issue that is suggested to user upon confirmation
+                if (!issuepickerQueryField.isDisabled()) issuepickerQueryField.appendText(";");
+                System.out.println(state.getSelectedIssues());
+                return state.getSelectedIssues().stream().map(TurboIssue::toString).collect(Collectors.toList());
+            }
+            return null;
+        });
+    }
+
+    private FlowPane createRepoPane(Insets padding) {
+        FlowPane repo = new FlowPane();
+        repo.setHgap(5);
+        repo.setVgap(5);
+        repo.setPadding(padding);
+        return repo;
+    }
+
+    private Label createRepoTitle(String name) {
+        Label repoName = new Label(name + ": ");
+        repoName.setPadding(new Insets(0, 5, 5, 0));
+        repoName.setMaxWidth(ELEMENT_MAX_WIDTH - 10);
+        repoName.setStyle("-fx-font-size: 110%; -fx-font-weight: bold; ");
+        return repoName;
+    }
+
+    private final Map<String, FlowPane> getRepoContent(List<TurboIssue> chosenIssues) {
+        Map<String, FlowPane> repoContent = new HashMap<>();
+        chosenIssues.stream()
```

Comment:
We can use forEach without the intermediate stream.

But anyway, this looks a like a mapping operation instead. We are transforming the list of chosen issues into a map of nodes.